onnxscript/rewriter/generic_pattern.py

```diff
@@ -294,6 +294,78 @@ def make_node(
         return proto
 
 
+class PatternMatchResult:
+    """Stores information about a match if a match was successful."""
+
+    def __init__(
+        self,
+        pattern: "GenericPattern",
+        model_nodes: typing.Sequence[oir.Node],
+        pattern_nodes: typing.Sequence[oir.Node],
+        pattern_input_names: typing.Sequence[str],
+        pattern_output_names: typing.Sequence[str],
+    ):
+        assert len(model_nodes) == len(pattern_nodes), (
+            f"The number of matched nodes in the model {len(model_nodes)} "
+            f"should the same as the number of nodes in the pattern "
+            f"{len(pattern_nodes)}."
+        )
+        self.pattern = pattern
+        self.model_nodes = model_nodes
+        self.pattern_nodes = pattern_nodes
+        self.pattern_input_names = pattern_input_names
+        self.pattern_output_names = pattern_output_names
+        self.kwargs = {}
+
+        matched_pattern_to_model_name: dict[str, str] = {}
```

Comment:
The other matcher uses a map from (pattern) names to `ir.Value`. I think that may be preferable (it will avoid an extra unnecessary lookup to convert the name into a `ir.Value` when users want to use it).

Reply:
The replacement needs this dictionary. It is more efficient in my case. The pattern names never change, the model names do.

Reply:
I don't understand. My point is not about pattern name and model names. The key used for the dictionary in both forms of MatchResult is the same (pattern name). But I am suggesting that the value returned by the dictionary can be an IR-value instead of an IR-value's name

Reply:
Created issue #1387